File: java/src/main/java/cn/leetcode/math/$67_AddBinary.java
package cn.leetcode.math;

public class $67_AddBinary {
    public String addBinary(String a, String b) {
        int carry = 0;
        char[] charArrayA = a.toCharArray();
        char[] charArrayB = b.toCharArray();
        StringBuilder str = new StringBuilder();
        for (int i = a.length() - 1, j = b.length() - 1; i >= 0 || j >= 0 || carry == 1; i--, j--) {
            int k1 = i <= -1 ? 0 : charArrayA[i] - '0';
            int k2 = j <= -1 ? 0 : charArrayB[j] - '0';
            int sum = k1 + k2 + carry;
            int re = sum % 2;
            carry = sum / 2;
            str.append(re);
        }
        return str.reverse().toString();
    }

    public static void main(String[] args) {
        String ans = new $67_AddBinary().addBinary("11", "1");
        System.out.println(ans);
    }
}

File: cpp/String/$67_Add Binary.cpp
class Solution {
public:
    cn.example.string addBinary(cn.example.string a, cn.example.string b) {
        cn.example.string res(max(a.size(),b.size()),'0');
        int i=a.size()-1,j=b.size()-1,k=max(a.size(),b.size())-1;
        int carry=0;
        while(i>=0 || j>=0){
            int aa= i>=0?a[i--]-'0':0;
            int bb= j>=0?b[j--]-'0':0;
            int sum=aa+bb+carry;
            if(sum>=2){//判断是否进位
                res[k--] = (sum % 2)+'0';
                carry=1;
            }else{
                res[k--]=sum+'0';
                carry=0;
            }
        }
        if(carry==1){//最高一位进位
            res='1'+res;
        }
        return res;
    }
};
